### PR TITLE
Dashboard templates & first default dashboard

### PR DESCRIPTION
**Pull request by paolodamico**
## Changes

Supersedes #1900 (please read original PR for full context).

- This PR introduces the capability to create static dashboard templates and ability to create dashboards from those templates.
- The PR also includes the app default dashboard referenced in #1694. _Please note this PR does not close the issue._ 

image

## Checklist

- [ ] All querysets/queries filter by Team (if this PR affects any querysets/queries)
- [ ] Backend tests (if this PR affects the backend)
- [ ] Cypress E2E tests (if this PR affects the front and/or backend)

### CODE REVIEW

**Comment by paolodamico** [comment]
Future improvements:
- Support for dynamic templates.
- Add backend tests for template usage.

**Comment by paolodamico** [comment]
A few things worth noting too:

- Maybe we ought to suggest a schema for some universal custom events (see [Segment spec](https://segment.com/docs/connections/spec/ecommerce/v2/) example), that way we can do a lot of more cool things (e.g. the dashboard would become "useful" automatically instead of needing some manual tweaks). @yakkomajuri POV?

- I think it'd be helpful to be able to change the legends of the funnel steps (this could help with the missing empty state too).

**Comment by yakkomajuri** [comment]
@paolodamico RE schema I've been thinking about this a bit after learning more about [Snowplow](https://snowplowanalytics.com/blog/2020/01/24/re-thinking-the-structure-of-event-data/). I think it's a use-case that fits plugins perfectly though, so I've been waiting on that. It could then be a plugin that's enabled by default or something

File: posthog/helpers/dashboard_templates.py [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+        name="Users by traffic source",
+        type=TRENDS_LINEAR,
+        filters={
+            TREND_FILTER_TYPE_EVENTS: [{"id": "$pageview", "math": "dau", "type": TREND_FILTER_TYPE_EVENTS}],
+            INTERVAL: "day",
+            BREAKDOWN_TYPE: "event",
+            BREAKDOWN: "$initial_referring_domain",
+        },
+        last_refresh=now(),
+    )
```
Comment (Twixes) [inline posthog/helpers/dashboard_templates.py:151]: Could maybe decouple dashboard item creation from the dry dashboard templates itself? Not super needed now, but would be more elegant if hydration was something like
```Python
DashboardItem.objects.bulk_create([
    DashboardItem(team=dashboard.team, last_refresh=now(), **item_schematic) for item_template in item_templaes
])
```

**Review by Twixes — APPROVED** [review]
LGTM. Side note: I agree wholeheartedly with Marius's enhancement proposals back in #1900.

**Pull request merged**